include/exadg/grid/grid_utilities.h

```diff
@@ -90,6 +90,86 @@ get_mesh_smoothing(bool const use_local_smoothing_multigrid, ElementType const &
   return mesh_smoothing;
 }
 
+/**
+ * This function can be seen as some form of "copy constructor" for periodic face pairs,
+ * transforming the template argument of dealii::GridTools::PeriodicFacePair from
+ * Triangulation::cell_iterator to DoFHandler::cell_iterator.
+ */
+template<int dim>
+std::vector<dealii::GridTools::PeriodicFacePair<typename dealii::DoFHandler<dim>::cell_iterator>>
+transform_periodic_face_pairs_to_dof_cell_iterator(
+  std::vector<dealii::GridTools::PeriodicFacePair<
+    typename dealii::Triangulation<dim>::cell_iterator>> const & periodic_faces,
+  dealii::Triangulation<dim> const &                             triangulation,
+  dealii::DoFHandler<dim> const &                                dof_handler)
+{
+  typedef typename std::vector<
+    dealii::GridTools::PeriodicFacePair<typename dealii::DoFHandler<dim>::cell_iterator>>
+    PeriodicFacesDoF;
+
+  PeriodicFacesDoF periodic_faces_dof;
+
+  for(auto it : periodic_faces)
+  {
+    dealii::GridTools::PeriodicFacePair<typename dealii::DoFHandler<dim>::cell_iterator>
+      face_pair_dof_hander;
+
+    face_pair_dof_hander.cell[0] = typename dealii::DoFHandler<dim>::cell_iterator(
+      &triangulation, it.cell[0]->level(), it.cell[0]->index(), &dof_handler);
+    face_pair_dof_hander.cell[1] = typename dealii::DoFHandler<dim>::cell_iterator(
+      &triangulation, it.cell[1]->level(), it.cell[1]->index(), &dof_handler);
+
+    face_pair_dof_hander.face_idx[0] = it.face_idx[0];
+    face_pair_dof_hander.face_idx[1] = it.face_idx[1];
+
+    face_pair_dof_hander.orientation = it.orientation;
+    face_pair_dof_hander.matrix      = it.matrix;
+
+    periodic_faces_dof.push_back(face_pair_dof_hander);
+  }
+
+  return periodic_faces_dof;
+}
+
+/**
+ * This function can be seen as some form of "copy constructor" for periodic face pairs,
+ * transforming the template argument of dealii::GridTools::PeriodicFacePair from
+ * DoFHandler::cell_iterator to Triangulation::cell_iterator.
+ */
+template<int dim>
+std::vector<dealii::GridTools::PeriodicFacePair<typename dealii::Triangulation<dim>::cell_iterator>>
+transform_periodic_face_pairs_to_tria_cell_iterator(
```

Comment:
Do we need the function, or could one even make the assignment to get an object of the type
```c++
  std::vector<dealii::GridTools::PeriodicFacePair<typename dealii::Triangulation<dim>::cell_iterator>>
```
by assigning it an object of the type
```c++
  std::vector<dealii::GridTools::PeriodicFacePair<typename dealii::DoFhandler<dim>::cell_iterator>>
```
This would be an upcast, so I would have assumed that the `std::vector` propagates through the automatic assignment operators.

Reply:
How would that work? The `DoFCellAccessor` needs a reference to the underlying `DoFHandler`!?

Reply:
> yes, not used currently, but I did it because it was very easy to obtain after I implemented the other function

I would remove it. I cannot think of an instance where one would need that kind of translation.

Reply:
I mean just the conversion `DoFHandler::cell_accessor` -> `Triangulation::cell_accessor` (currently not used), not the other way around.

Reply:
I will test the suggestion made here https://github.com/exadg/exadg/pull/388/files#r1137469215 (just out of curiosity), and remove this utility function, since you do not seem to see a value in that function if I got it right.

Reply:
> This would be an upcast, so I would have assumed that the std::vector propagates through the automatic assignment operators.

This does not seem to work!? Even if I do the assignment element by element (by a range based for-loop), the automatic assignment does not seem to work. (Note that there is a typo in the suggestion above: DoFhandler -> DoFHandler).

Should I remove the second utility function @kronbichler?

Reply:
Yes, I would remove this function. And for the record, I tested the automatic conversion as `vector_of_tria_iterators(vector_of_dofh_iterators.begin(), vector_of_dofh_iterators.end())` and realize that the `PeriodicFacePair` would need a custom constructor because the compiler would not automatically convert the iterators. So it would be more work than I suggested.